proxy/grpcproxy/watch_broadcasts.go

```diff
@@ -60,8 +60,10 @@ func (wbs *watchBroadcasts) coalesce(wb *watchBroadcast) {
 			continue
 		}
 		wbswb.mu.Lock()
-		// NB: victim lock already held
-		if wb.nextrev >= wbswb.nextrev && wbswb.nextrev != 0 {
+		// 1. check if wbswb is behind wb so it won't skip any events in wb
+		// 2. ensure wbswb started; nextrev == 0 may mean wbswb is waiting
+		// for a current watcher and expects a create event from the server.
+		if wb.nextrev >= wbswb.nextrev && wbswb.responses > 0 {
```

Comment:
It still fakes a creation event if the new watcher can be added to an existing watch broadcast. This is so that if a watcher has already seen a create, it won't try to join to a watch broadcast that is still waiting for its create event.